src/Sulu/Bundle/SecurityBundle/Resources/public/js/services/role-manager.js

```diff
@@ -0,0 +1,186 @@
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+define([
+    'services/husky/util',
+    'services/husky/mediator',
+    'sulusecurity/models/role'
+], function(
+    Util,
+    Mediator,
+    Role
+) {
+
+    'use strict';
+
+    var instance = null,
+
+        /**
+         * Saves a role with given data
+         *
+         * @param {Object} data The data to save
+         * @returns {Object} a promise
+         */
+        saveRole = function(data) {
+            var promise = $.Deferred(),
+                role = Role.findOrCreate({id: data.id});
+            role.set(data);
+
+            role.save(null, {
+                success: function(response) {
+                    promise.resolve(response.toJSON());
+                }.bind(this),
+                error: function(context, jqXHR) {
+                    promise.reject(jqXHR);
+                }.bind(this)
+            });
+
+            return promise;
+        },
+
+        /**
+         * Deletes a role for a given id
+         *
+         * @param roleId The id of the role to delete
+         * @returns {Object} a promise
+         */
+        deleteRole = function(roleId) {
+            var role = Role.findOrCreate({id: roleId}),
+                promise = $.Deferred();
+
+            role.destroy({
+                success: function() {
+                    Mediator.emit('sulu.security.role.deleted', roleId);
+                    Mediator.emit('sulu.labels.success.show', 'sulu-security.role.removed');
+                    promise.resolve();
+                }.bind(this),
+                error: function(context, jqXHR) {
+                    Mediator.emit('sulu.labels.error.show', getErrorStringForCode(jqXHR.responseJSON.code));
+                    promise.reject();
+                }.bind(this)
+            });
+
+            return promise;
+        },
+
+        /**
+         * For a given error code returns the corresponding
+         * error message.
+         *
+         * @param {Number} code The error code
+         * @returns {String} the corresponding error message
+         */
+        getErrorStringForCode = function(code) {
+            var translationKeyForError;
+            switch (code) {
+                case 1101:
+                    translationKeyForError = 'security.roles.error.non-unique';
+                    break;
+                default:
+                    translationKeyForError = '';
+                    break;
+            }
+
+            return translationKeyForError;
+        };
+
+    /** @constructor **/
+    function RoleManager() {
+    }
+
+    RoleManager.prototype = {
+
+        /**
+         * Load or create a role by given id
+         * @param {undefined|Number} roleId The role id to load the role for
+         * @returns {Object} a promise
+         */
+        loadOrNew: function(roleId) {
```

Comment:
I would prefer to know what the method is doing, so I would split this method into `load` and `create` or something like this. If a `create` is necessary at all, because we actually want to get rid of the backbone models.